sql/core/src/test/scala/org/apache/spark/sql/InjectRuntimeFilterSuite.scala

```diff
@@ -644,4 +644,76 @@ class InjectRuntimeFilterSuite extends QueryTest with SQLTestUtils with SharedSp
         "Missing or unexpected reused ReusedSubqueryExec in the plan")
     }
   }
+
+  test("Runtime bloom filter join: should add bf for left outer join even if left side is" +
+    " smaller than broadcast threshold") {
+    withSQLConf(SQLConf.RUNTIME_BLOOM_FILTER_APPLICATION_SIDE_SCAN_SIZE_THRESHOLD.key -> "600",
+      SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "300",
+      SQLConf.RUNTIME_BLOOM_FILTER_CREATION_SIDE_THRESHOLD.key -> "4000",
+      SQLConf.CBO_ENABLED.key -> "true") {
```

Comment:
Do we really need open CBO ?

Reply:
Yes it is required , as we use the size of join child to decided if one side can be broadcast.  In case of broadcast join, bloom filter is not applied.

Reply:
Even with CBO disabled, basic statistics are still propagated.

Reply:
+1. Please remove `SQLConf.CBO_ENABLED.key`.